editor/editor_inspector.cpp

```diff
@@ -3114,7 +3114,18 @@ void EditorInspector::update_tree() {
 			// Build the doc hint, to use as tooltip.
 
 			// Get the class name.
-			StringName classname = doc_name == "" ? object->get_class_name() : doc_name;
+			StringName classname = doc_name;
+			if (classname == "") {
```

Comment:
This should be a `else if` after the two checks below. There's no point doing all this `classname` computation before two cases which may overwrite it anyway.